components/engine_rocks/src/properties.rs

```diff
@@ -33,11 +39,32 @@ const PROP_RANGE_INDEX: &str = "tikv.range_index";
 pub const DEFAULT_PROP_SIZE_INDEX_DISTANCE: u64 = 4 * 1024 * 1024;
 pub const DEFAULT_PROP_KEYS_INDEX_DISTANCE: u64 = 40 * 1024;
 
-fn get_entry_size(value: &[u8], entry_type: DBEntryType) -> std::result::Result<u64, ()> {
+/// Caps the compaction factor to avoid unrealistic estimates.
+pub const TITAN_MAX_COMPACTION_FACTOR: f64 = 5000.0;
+const FIVE_MINS_IN_SECONDS: u64 = 5 * 60;
+
+lazy_static! {
+    pub static ref TITAN_COMPRESSION_FACTOR_SMOOTHER: Mutex<Smoother<f64, 30, FIVE_MINS_IN_SECONDS, 0>> =
+        Mutex::new(Smoother::<f64, 30, FIVE_MINS_IN_SECONDS, 0>::default());
+    pub static ref TITAN_COMPRESSION_FACTOR: AtomicU64 = AtomicU64::new(0);
```

Comment:
Although, it is guaranteed to be 1 in runtime. I feel that initialized it as 1 brings less confusion.